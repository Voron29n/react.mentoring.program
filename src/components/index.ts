export * from './common/Button';
export * from './common/Logo';

export * from './containers/Header';
export * from './containers/Lightbox';
export * from './containers/MovieForm';
export * from './containers/MovieForm/MovieFormView';
export * from './containers/MovieList';
export * from './containers/MovieList/MovieListView';

export * from './service/DeleteMovie';
export * from './service/EditMovie';
export * from './service/AddMovie';
export * from './service/SearchMovie';

export * from './single/Confirm';
export * from './single/Congratulations';
export * from './single/DropDown';
export * from './single/DropDown/DropDownItem';
export * from './single/Footer';
export * from './single/FilterBar';
export * from './single/GenreBar';
export * from './single/GenreBar/GenreBarItem';
export * from './single/GenreCheckmarks';
export * from './single/MovieDetail';
export * from './single/MovieItem/MovieView';
export * from './single/MovieItem';
export * from './single/MovieInput';
export * from './single/MovieInput/MovieInputView/index';
export * from './single/SearchBar';
export * from './single/SortBar';
export * from './single/Spinner';
export * from './single/Redirect';

export * from './error/ErrorBoundary';
export * from './error/FetchErrorMessage';
export * from './error/UnknownPage';
