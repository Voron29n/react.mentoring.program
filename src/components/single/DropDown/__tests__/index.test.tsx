import React from 'react';
import { render, RenderResult, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { DropDown } from 'components';
import {
  defaultClassnames,
  expectedClassnames,
  testClassnames,
  testDropdownList
} from './mocks';

const onClose = jest.fn();
const onSelected = jest.fn();

describe('DropDown component', () => {
  let dropDownRender: RenderResult;

  beforeEach(() => {
    dropDownRender = render(
      <DropDown
        classNames={testClassnames}
        selectList={testDropdownList}
        closeIcon={true}
        onClose={onClose}
        onSelected={onSelected}
      />
    );
  });

  it('renders', () => {
    const { container } = dropDownRender;
    const dropDownContainer = container.querySelector('.drop__down__container');
    const closeIcon = dropDownContainer.querySelector('.close__icon');

    expect(dropDownContainer).toBeInTheDocument();
    expect(dropDownContainer).toHaveClass(...expectedClassnames);

    expect(closeIcon).toBeInTheDocument();

    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getAllByRole('listitem').length).toBe(2);
  });

  it('onClose works', async () => {
    const { container } = dropDownRender;
    const closeIcon = container.querySelector('.close__icon').firstElementChild;
    await userEvent.click(closeIcon);
    expect(onClose).toHaveBeenCalledTimes(1);
  });

  it('onSelect works', async () => {
    await userEvent.click(screen.getAllByRole('listitem')[0]);
    expect(onSelected).toHaveBeenCalledTimes(1);
    expect(onSelected).toHaveBeenCalledWith(testDropdownList[0]);
  });

  it('DropDown snapshot', () => {
    expect(dropDownRender).toMatchSnapshot();
  });
});

describe('DropDown component with required props', () => {
  let dropDownRender: RenderResult;

  beforeEach(() => {
    dropDownRender = render(
      <DropDown selectList={testDropdownList} onSelected={onSelected} />
    );
  });

  it('renders', () => {
    const { container } = dropDownRender;
    const dropDownContainer = container.querySelector('.drop__down__container');
    const closeIcon = dropDownContainer.querySelector('.close__icon');

    expect(dropDownContainer).toBeInTheDocument();
    expect(dropDownContainer).toHaveClass(...defaultClassnames);

    expect(closeIcon).toBeNull();

    expect(screen.getByRole('list')).toBeInTheDocument();
    expect(screen.getAllByRole('listitem').length).toBe(2);
  });

  it('DropDown snapshot', () => {
    expect(dropDownRender).toMatchSnapshot();
  });
});
