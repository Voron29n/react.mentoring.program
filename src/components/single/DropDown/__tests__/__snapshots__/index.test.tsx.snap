// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown component DropDown snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="drop__down__container drop__down__test"
      >
        <div
          class="close__icon"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
            />
          </svg>
        </div>
        <ul>
          <li
            class="drop__down__item upper__text"
          >
            testLabel1
          </li>
          <li
            class="drop__down__item upper__text"
          >
            testLabel2
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="drop__down__container drop__down__test"
    >
      <div
        class="close__icon"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M289.94 256l95-95A24 24 0 00351 127l-95 95-95-95a24 24 0 00-34 34l95 95-95 95a24 24 0 1034 34l95-95 95 95a24 24 0 0034-34z"
          />
        </svg>
      </div>
      <ul>
        <li
          class="drop__down__item upper__text"
        >
          testLabel1
        </li>
        <li
          class="drop__down__item upper__text"
        >
          testLabel2
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DropDown component with required props DropDown snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="drop__down__container undefined"
      >
        <ul>
          <li
            class="drop__down__item upper__text"
          >
            testLabel1
          </li>
          <li
            class="drop__down__item upper__text"
          >
            testLabel2
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="drop__down__container undefined"
    >
      <ul>
        <li
          class="drop__down__item upper__text"
        >
          testLabel1
        </li>
        <li
          class="drop__down__item upper__text"
        >
          testLabel2
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
